cli/src/semgrep/core_runner.py

```diff
@@ -651,27 +653,29 @@ def plan_core_run(
             for language in rule.languages:
                 targets = list(
                     target_manager.get_files_for_rule(
-                        language,
-                        rule.includes,
-                        rule.excludes,
-                        rule.id,
+                        language, rule.includes, rule.excludes, rule.id, rule.product
                     )
                 )
 
                 for target in targets:
                     if all_targets is not None:
                         all_targets.add(target)
-                    target_info[target, language].append(rule_num)
+                    rules_nums, products = target_info[target, language]
+                    rules_nums.append(rule_num)
+                    products.add(rule.product.to_json_string())
```

Comment:
Why do we need `products` here? That can be indirectly obtained through the rule ids no?

Reply:
The rule numbers are no longer being passed to core. This is computing the set of products to pass to core. It can be computed anywhere where you have access to the list of rules, this is on such place.